### ISSUE TITLE
Incorrect Aggregate Calculation

### ISSUE THREAD

**Issue by lloydtabb** [duckdb/duckdb#3756]
The data in this test can be found by cloning 

```
git clone git@github.com:lloydtabb/imdb_malloy.git
make
```

It seems there is a bug in DISTINCTing.  Not sure exactly where.

```
SELECT
    group_set,
    COUNT(DISTINCT concat(movies_queries.tconst, '|', IFNULL(movies_queries.ratings.numVotes,0)::varchar) )  as all_movie_votes,
    case when group_set = 0 THEN
      COUNT(DISTINCT concat(movies_queries.tconst, '|', IFNULL(movies_queries.ratings.numVotes,0)::varchar) )
    END as movie_votes0,
    case when group_set = 1 THEN
      COUNT(DISTINCT concat(movies_queries.tconst, '|', IFNULL(movies_queries.ratings.numVotes,0)::varchar) )
    END as movie_votes1

  FROM 'titles.parquet' as movies_queries
  LEFT JOIN 'principals.parquet' AS principals_0
    ON movies_queries.tconst=principals_0.tconst
  -- comment out the line below to make the bug go away
  LEFT JOIN 'names.parquet' AS people_0 ON principals_0.nconst=people_0.nconst
  CROSS JOIN (SELECT UNNEST(GENERATE_SERIES(0,1,1)) as group_set  ) as group_set
  WHERE (movies_queries.ratings.numVotes>30000)
     -- comment the line below to make the bug go away.
     AND ((group_set NOT IN (1) OR (group_set IN (1) AND principals_0.category='director')))
  GROUP BY 1
  ```

The proper result is 6094 distinct movies.  There should be nothing a join can do to change the calculation of a count distinct, yet it does.  The where clause  should only effect calulations in group_set= 1, yet it effects calculations in group_set=0

Changing either of the lines below the comment will produce an accurate calculation.

![image](https://user-images.githubusercontent.com/1093458/171655646-5ebe92f3-2806-4877-ae12-7ea074550a77.png)

**Comment by lloydtabb** [comment]
Yes, it still happens on master...
![image](https://user-images.githubusercontent.com/1093458/171665746-7eb5b52b-6b69-4095-baa1-78648c1de1a6.png)

**Comment by Mytherin** [comment]
I believe I have tracked down the issue and fixed it in #3757, with those changes I get the following result for the query, which appears to be correct to me:


```sql
SELECT
    group_set,
    COUNT(DISTINCT concat(movies_queries.tconst, '|', IFNULL(movies_queries.ratings.numVotes,0)::varchar) )  as all_movie_votes,
    case when group_set = 0 THEN
      COUNT(DISTINCT concat(movies_queries.tconst, '|', IFNULL(movies_queries.ratings.numVotes,0)::varchar) )
    END as movie_votes0,
    case when group_set = 1 THEN
      COUNT(DISTINCT concat(movies_queries.tconst, '|', IFNULL(movies_queries.ratings.numVotes,0)::varchar) )
    END as movie_votes1

  FROM 'titles.parquet' as movies_queries
  LEFT JOIN 'principals.parquet' AS principals_0
    ON movies_queries.tconst=principals_0.tconst
  -- comment out the line below to make the bug go away
  LEFT JOIN 'names.parquet' AS people_0 ON principals_0.nconst=people_0.nconst
  CROSS JOIN (SELECT UNNEST(GENERATE_SERIES(0,1,1)) as group_set  ) as group_set
  WHERE (movies_queries.ratings.numVotes>30000)
     -- comment the line below to make the bug go away.
     AND ((group_set NOT IN (1) OR (group_set IN (1) AND principals_0.category='director')))
  GROUP BY 1
┌───────────┬─────────────────┬──────────────┬──────────────┐
│ group_set │ all_movie_votes │ movie_votes0 │ movie_votes1 │
├───────────┼─────────────────┼──────────────┼──────────────┤
│ 0         │ 6094            │ 6094         │ NULL         │
│ 1         │ 5186            │ NULL         │ 5186         │
└───────────┴─────────────────┴──────────────┴──────────────┘

```


Thanks again for the report!
